tests/test_environ.py

```diff
@@ -22,7 +24,12 @@ def test_environment_creation_preserves_PATH(testing_workdir, testing_config):
 
 def test_environment():
     """Asserting PrefixData can accomplish the same thing as Environment."""
-    assert (specs := environ.Environment(sys.prefix).package_specs())
+    with pytest.warns(
+        PendingDeprecationWarning
+        if deprecated._version_less_than("24.3")
```

Comment:
We should just use `packaging.version` as we do in other places and consider the version parsing in `DeprecationHandler` as purely internal, i.e., not use it outside of its scope.

Reply:
Fair enough

Reply:
See https://github.com/conda/conda-build/pull/5244/commits/6b026a1750207422de7eb56c2d6c49b36831c922